cmd/manager/main.go

```diff
@@ -68,125 +72,153 @@ const (
 )
 
 var (
-	// Cmd is the cobra command to start the manager.
-	Cmd = &cobra.Command{
+	configFile string
+	log        logr.Logger
+)
+
+func Command() *cobra.Command {
+	cmd := &cobra.Command{
 		Use:   "manager",
-		Short: "Start the operator manager",
-		Long: `manager starts the manager for this operator,
- which will in turn create the necessary controller.`,
-		Run: func(cmd *cobra.Command, args []string) {
-			execute()
+		Short: "Start the ECK operator",
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			if configFile != "" {
+				viper.SetConfigFile(configFile)
+				if err := viper.ReadInConfig(); err != nil {
+					return fmt.Errorf("failed to read config file %s: %w", configFile, err)
+				}
+
+				viper.WatchConfig()
+				viper.OnConfigChange(func(_ fsnotify.Event) {
+					if log != nil {
+						log.Info("Detected an update to the configuration file. Restart the operator for the changes to take effect.")
```

Comment:
This is likely to be missed by users though? Is it worth having it? Or can we make it more prominent with an event maybe?

Reply:
I mainly added it as an aid to jog the users memory. Hopefully, most users will look at the logs when they notice that the config change hasn't taken effect. Even if they don't, someone providing support will :crossed_fingers: 

I'll look into adding an event too but I suspect that it'll be tricky to do at such an early stage of the operator life cycle because the controller runtime hasn't been initialized yet.